src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8770,7 +8770,11 @@ internal static void DeleteFileStream(string path, string streamName)
             }
             string resultPath = path + adjustedStreamName;
 
-            NativeMethods.DeleteFile(resultPath);
+            if (!NativeMethods.DeleteFile(resultPath))
+            {
+                int error = Marshal.GetLastWin32Error();
+                throw new Win32Exception(error);
```

Comment:
Shouldn't this be a non-terminating error? The following should behave similarly:
```powershell
dir | Unblock-File
dir | Remove-Item
```
In the `Remove-Item` case, a read only file reports an error, but all other files are deleted.